finality-aleph/src/network.rs

```diff
@@ -233,24 +271,51 @@ enum MetaMessage {
     AuthenticationRequest(SessionId),
 }
 
+type MessageIndex = u64;
+
+#[derive(Clone, Encode, Decode)]
+struct DataMessage<D> {
+    session_id: SessionId,
+    index: MessageIndex,
+    data: D,
+    recipient: Recipient<NodeIndex>,
+}
+
+impl<D: Encode> Hash for DataMessage<D> {
+    fn hash<H: StdHasher>(&self, state: &mut H) {
+        self.index.hash(state);
+        self.data.encode().hash(state);
```

Comment:
Unlikely to surface, but I think this is technically a bug – if we have two nodes that want to send the same data, they  might assign it the same index and then one of the packets might be discarded. Since we cannot depend on the source of the data anywhere this can only be a problem if both packets are being sent to different destinations. Adding `recipient` to the hash would solve the problem.

It's quite possible this wouldn't be a problem for AlephBFT in practice, bu it's a strange limitation to have to remember.

Reply:
Wait, no, this is not a bug, because we already assume any packet might not reach it's destination for random reasons. Leaving above comment, but just informational.